### PR TITLE
Changing discourse emailer from SendGrid to MailGun

### PR DESCRIPTION
**Pull request by Maxrimus**
SendGrid has changed it's security features to require SMS based 2FA and as such we had to change our email sending service. I have changed us over to MailGun. The code has the same basic structure but changed to work properly with MailGun.

### CODE REVIEW

**Comment by leekillough** [comment]
I'd probably squash the 60+ commits into one. I usually do this with `git reset --soft master` (after a `git fetch; git merge master`), followed by `git commit` with one new message, although you can also do it with `git rebase -i HEAD~67` (or the like) if you want to preserve the commit comment history and edit it interactively.

GitHub has "squash and merge", but this is not a true squash and merge -- it simply creates a completely new commit on the target branch, with no history to its predecessor branch.

**Comment by e-kayrakli** [comment]
> although you can also do it with `git rebase -i HEAD~67` (or the like)

`git rebase -i master` should be a tiniest bit easier

**Comment by Maxrimus** [comment]
> I'd probably squash the 60+ commits into one. I usually do this with `git reset --soft master` (after a `git fetch; git merge master`), followed by `git commit` with one new message, although you can also do it with `git rebase -i HEAD~67` (or the like) if you want to preserve the commit comment history and edit it interactively.
> 
> GitHub has "squash and merge", but this is not a true squash and merge -- it simply creates a completely new commit on the target branch, with no history to its predecessor branch.

I went ahead and squashed the commits 👍

**Review by leekillough — APPROVED** [review]
If it works, go ahead.

File: emailer.py [hunk 1]
```diff
@@ -12,6 +10,8 @@
 import rollbar
 import rollbar.contrib.flask
 import sha
+import smtplib
```
Comment (ronawho) [inline emailer.py:13]: Do we need to update `requirements.txt` now? (Dropping `Envelopes` and adding whatever package has `smtplib`  )

Comment (ronawho) [reply emailer.py:13]: Ah, looks like `smtplib` is part of the standard library so probably don't need a new package, but could still drop envelopes?

File: emailer.py [hunk 2]
```diff
@@ -124,7 +111,7 @@ def _send_email(msg_info):
     """Create and send commit notification email."""
     sender = _get_sender(msg_info['pusher_email'])
     recipient = os.environ.get('GITHUB_COMMIT_EMAILER_RECIPIENT')
-
+
```
Comment (ben-albrecht) [inline emailer.py:114]: trailing whitespace

File: emailer.py (hunk 1 above)
Comment (Maxrimus) [reply emailer.py:13]: smptlib is a default python library. I could remove Envelopes for sure but I shouldn't have to add anything.

File: emailer.py [hunk 3]
```diff
@@ -38,21 +38,6 @@ def init_rollbar():
         rollbar.contrib.flask.report_exception, app)
 
 
-@app.before_request
-def app_before_request():
-    envelopes.connstack.push_connection(
-        envelopes.SendGridSMTP(
-            login=os.environ.get('SENDGRID_USERNAME'),
```
Comment (ronawho) [inline emailer.py:45]: Probably some documentation changes needed to the top-level README now (e.g. `SENDGRID_PASSWORD` -> `MAILGUN_PASSWORD`)

**Review by ben-albrecht — APPROVED** [review]
I have no experience with smtplib or mailgun, but the python code LGTM, nice work!

**Pull request merged**